pkg/storage/admitters/vmsnapshot.go

```diff
@@ -90,13 +89,7 @@ func (admitter *VMSnapshotAdmitter) Admit(ctx context.Context, ar *admissionv1.A
 
 		switch *vmSnapshot.Spec.Source.APIGroup {
 		case core.GroupName:
-			switch vmSnapshot.Spec.Source.Kind {
-			case "VirtualMachine":
-				causes, err = admitter.validateCreateVM(ctx, sourceField.Child("name"), ar.Request.Namespace, vmSnapshot.Spec.Source.Name)
-				if err != nil {
-					return webhookutils.ToAdmissionResponseError(err)
-				}
-			default:
+			if vmSnapshot.Spec.Source.Kind != "VirtualMachine" {
```

Comment:
Just a thought, the `switch` above could be also simplified if we want to clean-up nested switches. Not a must though.